libcfnet/protocol.h

```diff
@@ -64,4 +64,52 @@
  */
 Seq *ProtocolOpenDir(AgentConnection *conn, const char *path);
 
+/**
+ * Receives a file from a remote host.
+ *
+ * The server will use "/var/cfengine" as working directory if absolute path
+ * is not specified. The client will send a request that looks like this:
+ *      `GET <file_size> <remote_path>`
+ *
+ * `file_size` is the maximum size of the file to receive.
+ *
+ * The function shall fail if connection is not established, or if the server
+ * gives a bad response (denoted by a message preceded by "BAD").
+ *
+ * @param [in] conn         The connection to use
+ * @param [in] remote_path  Path on remote host
+ * @param [in] local_path   Path of received file
+ * @param [in] file_size    Size of file to get
+ * @param [in] perms        Permissions of local file
+ * @return True if file was successfully transferred, false otherwise
+ *
+ * Example (for printing each directory entry):
+ * @code
+ *     AgentConnection *conn = ServerConnection("127.0.0.1", "666", ...);
+ *     bool got_file = ProtocolGet(conn, "masterfiles/update.cf",
+ *                                 "update.cf", CF_MSGSIZE, 0644);
+ *     if (got_file)
+ *     {
+ *        struct stat sb;
+ *        stat("update.cf", &sb);
+ *        printf("This file is %ld big!\n", sb.st_size);
+ *     }
+ * @endcode
+ *
+ * In the protocol, this will look like this on the server side:
+ *     Received: GET masterfiles/update.cf
+ *     Translated to: GET /var/cfengine/masterfiles/update.cf
+ */
+bool ProtocolGet(AgentConnection *conn, const char *remote_path,
+                 const char *local_path, const uint32_t file_size, int mode);
```

Comment:
Should be named `perms` instead of `mode` to correspond with how we name things in `file_lib`, maybe